src/agent/src/image.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021, 2023 IBM Corporation
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use safe_path::scoped_join;
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use anyhow::{anyhow, bail, Context, Result};
+use image_rs::image::ImageClient;
+use tokio::sync::Mutex;
+
+use crate::rpc::CONTAINER_BASE;
+use crate::AGENT_CONFIG;
+
+// A marker to merge container spec for images pulled inside guest.
+const ANNO_K8S_IMAGE_NAME: &str = "io.kubernetes.cri.image-name";
+const KATA_IMAGE_WORK_DIR: &str = "/run/kata/image/";
```

Comment:
`/run/kata/image/` => `/run/kata-containers/image/` to consistent with other data saved by agent.